src/gmp_backend.rs

```diff
@@ -239,6 +236,13 @@ impl Bn {
     pub fn to_bytes(&self) -> Vec<u8> {
         debug_assert!(self >= &Self::zero());
 
+        (&self.0).into()
+    }
+
+    /// Convert this big number to a big-endian byte sequence
+    pub fn to_signed_bytes(&self) -> Vec<u8> {
```

Comment:
What does "signed" bytes mean?  I see that this function returns a zero vec if self is negative, so in what sense is it "signed"?

Perhaps you intended to follow this pattern of returning a `(Sign, Vec<u8>)` pair?
https://github.com/rust-num/num-bigint/blob/1fc25378c1caf8f97ce56da58ed8bc40a13e3e76/src/bigint.rs#L731

Also, why does this function still use hex string conversion?

Reply:
I kept the old method and just renamed it without thinking too deeply about it. The old method didn't actually work with negative big nums. So, maybe I should just remove this method?

Reply:
I vote to remove it.  It should never be used, right?

Reply:
Yeah, we don't need it since we shouldn't get negative values.